spec/components/email/sender_spec.rb

```diff
@@ -58,6 +58,12 @@
     Email::Sender.new(message, :hello).send
   end
 
+  it "doesn't deliver when the to address uses the .invalid tld" do
+    message = Mail::Message.new(body: 'hello', to: 'myemail@example.invalid')
+    message.expects(:deliver_now).never
+    Email::Sender.new(message, :hello).send
```

Comment:
I think we should also expect that a `SkippedEmailLog` record was created with the reason `:sender_message_to_invalid`